qovery/resource_job.go

```diff
@@ -0,0 +1,511 @@
+								},
+							},
+						}),
+					},
+					"cronjob": {
+						Description: "Job's cron.",
+						Optional:    true,
+						Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+							"schedule": {
+								Description: "Job's cron string.",
+								Type:        types.StringType,
+								Required:    true,
+								// TODO(benjaminch): introduce a cron string validator
+							},
+							"command": {
+								Description: "Job's cron command.",
+								Required:    true,
+								Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+									"entrypoint": {
+										Description: "Entrypoint of the job.",
+										Type:        types.StringType,
+										Optional:    true,
+									},
+									"arguments": {
+										Description: "List of arguments of this job.",
+										Optional:    true,
+										Type: types.ListType{
+											ElemType: types.StringType,
+										},
+										PlanModifiers: tfsdk.AttributePlanModifiers{
+											modifiers.NewStringSliceDefaultModifier([]string{}),
+										},
+									},
+								}),
+							},
+						}),
+					},
+				}),
+			},
+			"source": {
+				Description: "Job's source.",
+				Optional:    true,
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+					"image": {
+						Description: "Job's image source.",
+						Optional:    true,
+						Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+							"registry_id": {
+								Description: "Job's image source registry ID.",
+								Type:        types.StringType,
+								Required:    true,
+							},
+							"name": {
+								Description: "Job's image source name.",
+								Type:        types.StringType,
+								Required:    true,
+							},
+							"tag": {
+								Description: "Job's image source tag.",
+								Type:        types.StringType,
+								Required:    true,
+							},
+						}),
+					},
+					"docker": {
+						Description: "Job's docker source.",
+						Optional:    true,
+						Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+							"dockerfile_path": {
+								Description: "Job's docker source dockerfile path.",
+								Type:        types.StringType,
+								Optional:    true,
+							},
+							"git_repository": {
+								Description: "Job's docker source git repository.",
+								Required:    true,
+								Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+									"url": {
+										Description: "Job's docker source git repository URL.",
+										Type:        types.StringType,
+										Required:    true,
+									},
+									"branch": {
+										Description: "Job's docker source git repository branch.",
+										Type:        types.StringType,
+										Required:    true,
+									},
+									"root_path": {
+										Description: "Job's docker source git repository root path.",
+										Type:        types.StringType,
+										Optional:    true,
```

Comment:
When playing with this one, I encountered a state issue (missing computed at first sight)
![image](https://user-images.githubusercontent.com/124068000/230119032-2625cae5-5aad-4c1c-9403-7ae22c617373.png)

Reply:
Do you have the value you set?

Reply:
Added `Computed:    true,`, should fix it IMO, let's try

Reply:
Ah sorry yes, I just didn't set the `rootè_path`:
```
git_repository = {
          url = "https://github.com/Qovery/terraform-provider-testing.git"
          branch = "job-echo-n-seconds"
        }
```

Reply:
It looks like the value is set to `null` in the terraform state

Reply:
Checking

Reply:
Fixed